### ISSUE TITLE
Ollama running, but not visible to Perplexica backend

### ISSUE THREAD

**Issue by Adrianogba** [ItzCrazyKns/Perplexica#535]
**To Reproduce**
The steps I did on my windows machine:
Clone the project
modify the `docker-compose.yaml` to add ollama:
```
services:
  searxng:
    image: docker.io/searxng/searxng:latest
    volumes:
      - ./searxng:/etc/searxng:rw
    ports:
      - 4000:8080
    networks:
      - perplexica-network
    restart: unless-stopped

  perplexica-backend:
    build:
      context: .
      dockerfile: backend.dockerfile
    image: itzcrazykns1337/perplexica-backend:main
    environment:
      - SEARXNG_API_URL=http://searxng:8080
      - OLLAMA_API_URL=http://ollama:11434
    depends_on:
      - searxng
      - ollama
    ports:
      - 3001:3001
    volumes:
      - backend-dbstore:/home/perplexica/data
      - uploads:/home/perplexica/uploads
      - ./config.toml:/home/perplexica/config.toml
    extra_hosts:
      - 'host.docker.internal:host-gateway'
    networks:
      - perplexica-network
    restart: unless-stopped

  perplexica-frontend:
    build:
      context: .
      dockerfile: app.dockerfile
      args:
        - NEXT_PUBLIC_API_URL=http://127.0.0.1:3001/api
        - NEXT_PUBLIC_WS_URL=ws://127.0.0.1:3001
    image: itzcrazykns1337/perplexica-frontend:main
    depends_on:
      - perplexica-backend
    ports:
      - 3000:3000
    networks:
      - perplexica-network
    restart: unless-stopped

  ollama:
    image: ollama/ollama:latest
    environment:
      - OLLAMA_HOST=0.0.0.0
      - OLLAMA_MODEL_PATH=/models
    volumes:
      - ollama-models:/models
    ports:
      - 11434:11434
    networks:
      - perplexica-network
    restart: unless-stopped

networks:
  perplexica-network:

volumes:
  backend-dbstore:
  uploads:
  ollama-models:
```
rename the `sample.config.toml` to `config.toml`
modify the OLLAMA with values such as `http://localhost:11434` or `http://host.docker.internal:11434`
```
[GENERAL]
PORT = 3001 # Port to run the server on
SIMILARITY_MEASURE = "cosine" # "cosine" or "dot"
KEEP_ALIVE = "5m" # How long to keep Ollama models loaded into memory. (Instead of using -1 use "-1m")

[API_KEYS]
OPENAI = "" # OpenAI API key - sk-1234567890abcdef1234567890abcdef
GROQ = "" # Groq API key - gsk_1234567890abcdef1234567890abcdef
ANTHROPIC = "" # Anthropic API key - sk-ant-1234567890abcdef1234567890abcdef
GEMINI = "" # Gemini API key - sk-1234567890abcdef1234567890abcdef

[API_ENDPOINTS]
SEARXNG = "http://localhost:32768" # SearxNG API URL
OLLAMA = "http://localhost:11434" # Ollama API URL - http://host.docker.internal:11434
```

**Screenshots**
![image](https://github.com/user-attachments/assets/d642bd65-ef3b-4e3f-8cfe-55173ed212c7)
![image](https://github.com/user-attachments/assets/8a132684-ad49-421a-b526-740f58a3ebf6)
![image](https://github.com/user-attachments/assets/db3c886c-a94c-4cb9-98c9-1a7ce8514587)

**Additional context**
Didn't modified nothing else, my objective is just to run it with Ollama and it seems everything is running, just that the backend of perplexica doesn't recognize the ollama image/service.

**Comment by ItzCrazyKns** [comment]
Try using `http://ollama:11434` as the Ollama address in the setttings.

**Comment by Adrianogba** [comment]
I did, but still doesn't work.
I deleted every volume and containers, modified the address to `OLLAMA = "http://ollama:11434"` runned `docker-compose up --build -d` but the result is the same.
The `docker-compose.yaml` is as above.
I think there is something on the docker file I'm not seeing or some configuration that might be missing.
![image](https://github.com/user-attachments/assets/8eaa5499-047b-4c70-9ad0-5bf4aee4707e)
I might try to pay some service to upload if the issue might be my pc, because this is the only pc I have and I can't install another OS to test this.
What you think might be the issue @ItzCrazyKns ? Or what you recommend to be done differently?

**Comment by Adrianogba** [comment]
@ItzCrazyKns I got it working, the address tip you gave was it.
But what got me confuse was that the error remained the same. Looking on the logs I saw that this time there was no 404 error on the perplexica backend (suggesting the communication is working), but a request on `/api/tags` on ollama and it's return was empty (no model was downloaded, since it was a fresh install). Then I run `ollama pull gemma2` on the container, configured that as the model on the configuration on the front and it worked!
